src/Subscriber/PaymentSubscriber.php

```diff
@@ -380,58 +401,49 @@ private function trans(string $snippet, array $parameters = []): string
      * Persists the Adyen payment state data on payment method confirmation/update
      * @param SalesChannelContextSwitchEvent $event
      */
-    private function saveStateData(SalesChannelContextSwitchEvent $event)
+    private function saveStateData(SalesChannelContextSwitchEvent $event, PaymentMethodEntity $selectedPaymentMethod)
     {
         //State data from the frontend
         $stateData = $event->getRequestDataBag()->get('adyenStateData');
 
-        if ($stateData) {
-            //Convert the state data into an array
-            $stateDataArray = json_decode($stateData, true);
+        //Convert the state data into an array
+        $stateDataArray = json_decode($stateData, true);
 
-            if (json_last_error() !== JSON_ERROR_NONE) {
-                $this->logger->error('Payment state data is an invalid JSON: ' . json_last_error_msg());
-                $this->session->getFlashBag()
-                    ->add('danger', $this->trans('adyen.paymentMethodSelectionError'));
-                return;
-            }
-
-            //Payment method selected from the Shopware methods form
-            $selectedPaymentMethod = $this->paymentMethodRepository->search(
-                (new Criteria())
-                    ->addFilter(new EqualsFilter('id', $event->getRequestDataBag()->get('paymentMethodId'))),
-                Context::createDefaultContext()
-            )->first();
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            $this->logger->error('Payment state data is an invalid JSON: ' . json_last_error_msg());
+            $this->session->getFlashBag()
```

Comment:
@peterojo Using the session flash bag here means that the store API will not be notified about it. maybe you can use the validation for it like the switch route itself does?

If not, I guess it isn't that big an issue if the store api does not receive feedback.

Reply:
Indeed, at the moment only we have the logs that the storeAPI client can use for debugging, but I am investigating what it takes to add store API responses for all our error messages throughout the plugin

Reply:
@peterojo Ah. Thanks. as mentioned not that big an issue anyway.